### PR TITLE
[ASCII-2516] use remote tagger for security and trace agent

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Ensure the trace and security agent use the `remote-tagger` and removes the configuration that allow to switch from local to remote.

### Motivation

Help with the quality gates incident https://app.datadoghq.com/incidents/32196

By removing the use of dual-tagger we can get rid of a few dependencies on the trace and security agent

### Describe how to test/QA your changes

We have being using the remote tagger for trace and security agent for the last three years https://github.com/DataDog/datadog-agent/pull/9842. 

I'm not sure if the test cover the cases for remote tagger, if it is not the case I would ask the @DataDog/agent-apm and @DataDog/agent-security to provide the necessary QA instructions

### CODE REVIEW

File: releasenotes/notes/use-remote-tagger-for-trace-and-security-agents-9650d1be5d49e47f.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+deprecations:
+  - |
+    ``apm_config.remote_tagger`` and ``security_agent.remote_tagger`` are deprecated and removed.
```
Comment (pgimalac) [inline releasenotes/notes/use-remote-tagger-for-trace-and-security-agents-9650d1be5d49e47f.yaml:11]: **💭 thought:** ‏The configurations are not documented (not part of the config template) so it seems odd to make a release note to announce that they are removed

**Review by pgimalac — APPROVED** [review]
Very nice results, LGTM !

File: pkg/security/probe/config/config.go [hunk 2]
```diff
@@ -171,7 +168,6 @@ func NewConfig() (*Config, error) {
 		ERPCDentryResolutionEnabled:  getBool("erpc_dentry_resolution_enabled"),
 		MapDentryResolutionEnabled:   getBool("map_dentry_resolution_enabled"),
 		DentryCacheSize:              getInt("dentry_cache_size"),
-		RemoteTaggerEnabled:          getBool("remote_tagger"),
```
Comment (paulcacheux) [inline pkg/security/probe/config/config.go:174]: can you remove the bind side as well ?

File: releasenotes/notes/use-remote-tagger-for-trace-and-security-agents-9650d1be5d49e47f.yaml (hunk 1 above)
Comment (GustavoCaso) [reply releasenotes/notes/use-remote-tagger-for-trace-and-security-agents-9650d1be5d49e47f.yaml:11]: I found a reference on the `CHANGELOG.rst` https://github.com/DataDog/datadog-agent/blob/a8b89968b48e3b5288d0ae5e6796447445dcec45/CHANGELOG.rst#L6199

**Review by paulcacheux — APPROVED** [review]

File: releasenotes/notes/use-remote-tagger-for-trace-and-security-agents-9650d1be5d49e47f.yaml (hunk 1 above)
Comment (paulcacheux) [reply releasenotes/notes/use-remote-tagger-for-trace-and-security-agents-9650d1be5d49e47f.yaml:11]: can you also mention the system-probe flag here ?

**Review by brycekahle — APPROVED** [review]

**Review by cswatt — APPROVED** [review]

**Review by ichinaski — APPROVED** [review]

File: releasenotes/notes/use-remote-tagger-for-trace-and-security-agents-9650d1be5d49e47f.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+deprecations:
+  - |
+    ``apm_config.remote_tagger``,  ``security_agent.remote_tagger``, and ``event_monitoring_config.remote_tagger`` config entries are deprecated and removed.
```
Comment (ajgajg1134) [inline releasenotes/notes/use-remote-tagger-for-trace-and-security-agents-9650d1be5d49e47f.yaml:11]: Maybe we should be a bit more explicit that the config here is now that the remote_tagger is always in use (so setting this config to false has no effect)

**Pull request merged**